### ISSUE TITLE
Bash won't complete install properly

### ISSUE THREAD

**Issue by csar** [Microsoft/BashOnWindows#718]
Please use the following bug reporting template to help produce actionable and reproducible issues:

Install bash hangs when "init" is executing using a full core but no visible activity

You can "successfully" end the installation, but then each attempt to run bash will just result in init spinning forever

![image](https://cloud.githubusercontent.com/assets/16315/17350821/2575a126-592a-11e6-8633-b2cac8af33ae.png)

**Comment by JacobStearns** [comment]
I'm having the exact same issue. I've gone through the correct steps to install Bash. I'm 100% sure about that. The install hangs at "Extracting filesystem" with the init process showing up in task manager. I've tried installing via "bash" in the start menu as well as with `lxrun /install` in both cmd and powershell.

**Comment by Xazax-hun** [comment]
I am also experiencing the exact same issue.

**Comment by atlgeek007** [comment]
I am also experiencing this issue.  I left my command prompt at "Extracting filesystem..."" overnight and it did not complete.

**Comment by onomatopellan** [comment]
Did any of you made a clean install of Anniversary Update?
Are you guys running any antivirus/firewall that is not Windows Defender?
Did you ever change/move/symlink LocalAppData or AppData folders to another partitions?

**Comment by Xazax-hun** [comment]
I did a windows update (from a previously fully updated windows 10 version), not a clean install.
I do have Avast antivirus, but I also tried it with the antivirus software disabled, without success.
I did not do any change to LocalAppData or AppData, they are at the default location.

**Comment by sauravMSFT** [comment]
I have run into the same problem. Previously I tried installing Bash on Ubuntu on Windows using the Bash command in Command Prompt. The installation got stuck at "Extracting filesystem, this will take a few minutes..." 
then I had to uninstall it using
```cmd
lxrun /uninstall
```
but only after a restart, as the bash process was still running and killing it from PowerShell gave Access Denied exception. I only have Windows Defender as my antivirus program.

**Comment by atlgeek007** [comment]
Relatively fresh install of Windows 10 Pro, windows update installed the Anniversary Update yesterday, which necessitated a series of reboots.

I have disabled all antivirus protection (Avast) and turned off malwarebytes also.

LocalAppData and AppData are not symlinked and in their default locations.

While it says "extracting filesystem" there is an "init" process owned by my user taking 10-20% CPU time.

**Comment by atlgeek007** [comment]
Oh, and I'm joined to a test domain, but I am logged in to an account with local administrator and domain administrator access to test this.

**Comment by JacobStearns** [comment]
I didn't do a clean install, just an update. I'm running Avast, but I've tried disabling it without any luck. As for messing with AppData and LocalAppData, I haven't done anything to them.

My init process is also running at about 10-20%, but nothing happens.

**Comment by csar** [comment]
Avast here, I also tried to deactivate it, no change.
I never used a developer preview. I will try on another machine later.
One thing might have caused trouble : l had gitbash installed, so there was
already a bash. Exe

**Comment by onomatopellan** [comment]
A RamDisk or Virtual drive could affect the install, disable them if you have one.

There is a known problem with having more than one pagefile. https://github.com/Microsoft/BashOnWindows/issues/565

It could also be related to this new unsolved bug. https://github.com/Microsoft/BashOnWindows/issues/707

**Comment by JacobStearns** [comment]
I've only got one pagefile and I don't have any virtual drives or RAM disks set up, so that shouldn't be an issue on my end.

**Comment by onomatopellan** [comment]
It seems #707 is now solved. Did you try looking for the reg entry?`HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\kernel\ dword:obcaseinsensitive` should be `1`

**Comment by csar** [comment]
I just updated a Win 10 Pro machine and everything went smoothly, I didn't even had to enable developer mode - it did it for me. Is there an issue with Home edition

**Comment by JacobStearns** [comment]
It was already set to 1, yeah. I should note that there is not an error message like in #707. The install just hangs indefinitely.

**Comment by onomatopellan** [comment]
If you WIn+R and go to %localappdata%/lxss what do you see? Can you enter on rootfs directory and see its contents?

**Comment by JacobStearns** [comment]
I can enter lxss/rootfs, yes. It contains multiple folders as well as the "init" file.

**Comment by onomatopellan** [comment]
If you didn't see any lxxs.tar.gz file on lxss folder then it means it extracted the files correctly, deleted the lxxs.tar.gz file and then it hangs.
Open an Admin CMD and type `lxrun /setdefaultuser root`
then type `bash`
Can you see any error?

**Comment by benhillis** [comment]
@obcaseinsensitive thanks for helping folks out.  To clarify WSL does not require that registry key.  We enable case sensitivity temporarily on a per-thread basis for pico processes that are running Linux binaries.

@JacobStearns do you happen to have a third party firewall installed?  We've seen the install hang for users using Kaspersky and Bitdefender firewalls because they are blocking the network traffic of our pico processes.

**Comment by JacobStearns** [comment]
@onomatopellan Just to confirm, there was no lxxs.tar.gz, so that should all be fine.

I'm running `lxrun /setdefaultuser root`, but it seems to hang as before with the init process running at ~13%.

@benhillis I do not. I'm running Avast free, but that doesn't contain a firewall. I've tried disabling Avast, but that doesn't seem to help at all.

**Comment by salvavn** [comment]
@JacobStearns, I have exactly the same issue and the same behaviour after executing the` lxrun /setdefaultuser root` command. Nothing happens.

And I have the same content than you in the rootfs directory. AND I have running (but disabled) the free version of Avast...

Pretty similar.

![screenshot](https://cloud.githubusercontent.com/assets/12507762/17371491/59e8c7d2-59a0-11e6-9119-8c160db64898.PNG)

![screenshot2](https://cloud.githubusercontent.com/assets/12507762/17371520/7308a250-59a0-11e6-9440-1f6ad9b326ee.PNG)

**Comment by salvavn** [comment]
BTW, I have only one pagefile

![screenshot3](https://cloud.githubusercontent.com/assets/12507762/17371610/c9c605f6-59a0-11e6-9ce1-b61754703c3a.PNG)

Another question, do you have some kind of software like NoMachine Nx server installed?

**Comment by JacobStearns** [comment]
@salvavn Can confirm, this is the exact behavior I'm seeing.

**Comment by JacobStearns** [comment]
@salvavn I do not.

**Comment by salvavn** [comment]
It seems like lots of guys with the same issue have Avast working... I don't know if it's only a coincidence...

**Comment by onomatopellan** [comment]
Could you try uninstalling Avast, restarting Windows and then trying bash again?

**Comment by JacobStearns** [comment]
Can do. I'll be back once that's done.

**Comment by salvavn** [comment]
Success!

I have removed Avast. Then I have uninstalled bash with  `lxrun /uninstall /full`.

After that, I have rebooted Windows. And then, I have installed the linux stack with: `lxrun /install`

And it works! 

Now the prompt has asked me for the user name and password.

**Comment by salvavn** [comment]
I'm going to install Avast again in order to check if there is some kind of interaction after the bash installation.

**Comment by salvavn** [comment]
Hey guys, bad news. 

I was executing in my bash prompt the `sudo apt-get update` while I was reinstalling AVAST. When the installation was completed the sources listing of Ubuntu stopped, and now, I can't execute bash another time, I get the same result of this issue. 

![screenshot4](https://cloud.githubusercontent.com/assets/12507762/17372428/21f221a8-59a4-11e6-9d82-19bc88c98f6f.PNG)

So, in my opinion, the current AVAST version (the latest one) is incompatible with bash.

:(

**Comment by JacobStearns** [comment]
Uninstalling Avast fixed it on my end as well. It's unfortunate that Avast seems to completely break Bash.

**Comment by salvavn** [comment]
I have tried disabling Avast, but didn't work. The only way I'm able to work with bash is with a full  deinstallation of Avast :(

**Comment by mihailim** [comment]
I can confirm the problem is Avast (unfortunately, I found this issue only after spending a couple of hours before figuring it out myself...)

The problem also occurs with Avast's free Business Antivirus version.

It's not enough to turn Avast's shields off; the product *has* to be uninstalled.

**Comment by onomatopellan** [comment]
It seems Firewalls and AV will have to be updated to support the new picoprocesses.

**Comment by csar** [comment]
Anyone having recommendations for AVs that are compatible with Bash? Webroot seems to work just fine. In fact I am pretty annoyed by Avast because it broke things because of false positives, adding a message to each outgoing mail ... good opportunity to ditch Avast for good

**Comment by denru01** [comment]
Disabling the AVs may not be enough. Uninstalling it is a good way, but how about "closing" the AVs temporarily?
I solved my problem by "closing" Kaspersky but "disabling" does not work.

**Comment by atlgeek007** [comment]
I had to uninstall Avast for it to work at all.  As soon as I reinstalled Avast after installing bash, I couldn't actually run bash anymore.

I was in the market for a new AV anyway, may as well test out Defender.

**Comment by Hades1996** [comment]
@salvavn It worked like a charm, uninstalling avast did the trick

Thanks

**Comment by benhillis** [comment]
Glad to hear.  To keep everybody in the loop I've been reaching out to third party firewall providers and will work with them on getting this fixed.

Thanks everyone for trying WSL!

**Comment by degoya** [comment]
here is another issue who deals with 3rd party firewalls.
https://github.com/Microsoft/BashOnWindows/issues/475

**Comment by WRyan78** [comment]
Thanks for working on this.
Same issue here and I also have avast installed.

**Comment by sauravMSFT** [comment]
After reinstalling Bash a couple of times...
I simply ran
```cmd
lxrun /install
```
and waited till it hung at the extraction stage. After a couple of hours. I simply closed the command prompt. Task manager showed bash running. The lxss folder didn't have the tarball, which probably means everything was extracted correctly. 

I realized I had multiple page files. So I set it to "Automatically manage paging file size for all drives", then rebotted. I also disabled Windows Firewall and turned real time protection off in Windows Defender. I do not have any other Antivirus installed, Then ran 
```cmd
lxrun /setdefaultuser root
```
It worked. I then created a new for me
However, I cannot find the 'Bash on Windows on Ubuntu' program in my Start Menu. I have to start a prompt or PS and run bash.exe to run it.

**Comment by onomatopellan** [comment]
Glad it worked for you.
If you can't find the tile look at this https://github.com/Microsoft/BashOnWindows/issues/696

**Comment by BKNR** [comment]
To me it seems that the current newest version of Avast Free Antivirus (12.3.2280 / build 12.3.2154.0) does not have this issue. Installing WSL hanged on the 'Extracting filesystem' part, but then I updated my Avast av and it got past that to a successful installation.

**Comment by JacobStearns** [comment]
It's working on my end, too. Bash and every program I've tested have worked with Avast so far. If there's any issues still present, I haven't found them.

**Comment by pipirt** [comment]
Hi all, I just wanted to give my two cents on the Avast business. Bash.exe had been working perfectly fine for me and has with minimal problems since it was first released. A little while ago, I tried to move a big directory into another directory and it got hung up. It remained hung up after trying to Ctrl-Z it into the background or Ctrl-C it out of existence. So I closed it out. Opening it again, it remained hung up. Restarted; still hung up.

Ultimately what appears to have fixed it is **permanently turning off Avast's File System Shield and then restarting**. After the reboot, bash became unhung! Then I could just turn on the File System Shield again.

Avast Settings > Components > File System Shield Off > Stop Permanently

TL;DR
Before completely uninstalling Avast, try permanently turning off File System Shield and then restarting.

**Comment by n3rd4i** [comment]
Avast supports exclusion path, maybe adding the lxss path:
 - C:\Users\<username>\AppData\Local\lxss\*

Maybe this can stop Avast messing up files for WSL ?

**Comment by jcollum-cambia** [comment]
Having a similar issue here. No Avast involved though. Tried to install bash and it hung on `Extracting filesystem...`. Waited about 15 mins (no progress indicator! lame) and then decided to kill it. Later ran ` lxrun /uninstall` from an elevated PS. That appears to be hung as well. There are about 7 folders in `AppData\Local\lxss`. 10 mins later there were still the same number of folders, so that uninstall doesn't appear to be uninstalling. 

I miss my mac.

**Comment by benhillis** [comment]
This should be resolved in Creator's Update. If others are experiencing this same issue in Creators Update or later, please let me know!

**Comment by raouls555** [comment]
having the same problem here. im not running Avast but have something called f-secure (it's called kpn veilig idk why). tried to run bash anyway. it worked before i used any commands. i used ls and the system freaked out and then outputted the right thing for some reason. the file location seems bugged to but but works anyway. it seems like it now only has weird visual bugs. i hope this information can help find where the installation stops.
